src/services/votes.ts

```diff
@@ -8,102 +8,217 @@ import { insertVotesSchema } from '../types';
 import { CycleStatusType } from '../types/cycles';
 import { z } from 'zod';
 
-export function saveVote(dbPool: PostgresJsDatabase<typeof db>) {
+export function getVotes(dbPool: PostgresJsDatabase<typeof db>) {
   return async function (req: Request, res: Response) {
     const userId = req.session.userId;
-    req.body.userId = userId;
+    const cycleId = req.params.cycleId;
+
+    if (!cycleId) {
+      return res.status(400).json({
+        errors: [
+          {
+            message: 'Expected cycleId in query params',
+          },
+        ],
+      });
+    }
 
-    // Query num_of_votes and user_id for a specific option_id
-    const queryQuestionOption = await dbPool.query.questionOptions.findFirst({
-      where: eq(db.questionOptions.id, req.body.optionId),
-    });
+    const votesRow = await getVotesForCycleByUser(dbPool, userId, cycleId);
 
-    req.body.questionId = queryQuestionOption?.questionId;
-    const body = insertVotesSchema.safeParse(req.body);
+    return res.json({ data: votesRow });
+  };
+}
 
-    if (!body.success) {
-      return res.status(400).json({ errors: body.error.issues });
-    }
+export async function getVotesForCycleByUser(
+  dbPool: PostgresJsDatabase<typeof db>,
+  userId: string,
+  cycleId: string,
+) {
+  const response = await dbPool.query.cycles.findMany({
+    with: {
+      forumQuestions: {
+        with: {
+          questionOptions: {
+            with: {
+              votes: {
+                where: and(eq(db.votes.userId, userId)),
+                limit: 1,
+                orderBy: [desc(db.votes.createdAt)],
+              },
```

Comment:
Is it possible to do something like this: 

```
select: { 
        max: db.votes.createdAt,
        where: { userId: userId } 
}
```

The orderBy with limit combination works to get the most recent vote by user but it is extremely hard to read. Usually limit just returns any vote and just do to the fact that it is combined with orderBy it will return the most recent one. The suggestion that I made above would be more explicit.